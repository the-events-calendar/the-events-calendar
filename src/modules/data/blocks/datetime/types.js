/**
 * Internal dependencies
 */
import { PREFIX_EVENTS_STORE } from '@moderntribe/events/data/utils';

export const SET_START_DATE_TIME = `${ PREFIX_EVENTS_STORE }/SET_START_DATE_TIME`;
export const SET_END_DATE_TIME = `${ PREFIX_EVENTS_STORE }/SET_END_DATE_TIME`;
export const SET_START_TIME = `${ PREFIX_EVENTS_STORE }/SET_START_TIME`;
export const SET_END_TIME = `${ PREFIX_EVENTS_STORE }/SET_END_TIME`;
export const SET_DATE_RANGE = `${ PREFIX_EVENTS_STORE }/SET_DATE_RANGE`;

export const SET_START_TIME_INPUT = `${ PREFIX_EVENTS_STORE }/SET_START_TIME_INPUT`;
export const SET_END_TIME_INPUT = `${ PREFIX_EVENTS_STORE }/SET_END_TIME_INPUT`;

export const SET_NATURAL_LANGUAGE_LABEL = `${ PREFIX_EVENTS_STORE }/SET_NATURAL_LANGUAGE_LABEL`;

export const SET_MULTI_DAY = `${ PREFIX_EVENTS_STORE }/SET_MULTI_DAY`;

export const SET_ALL_DAY = `${ PREFIX_EVENTS_STORE }/SET_ALL_DAY`;

export const SET_SEPARATOR_DATE = `${ PREFIX_EVENTS_STORE }/SET_SEPARATOR_DATE`;
export const SET_SEPARATOR_TIME = `${ PREFIX_EVENTS_STORE }/SET_SEPARATOR_TIME`;

export const SET_TIME_ZONE = `${ PREFIX_EVENTS_STORE }/SET_TIME_ZONE`;
export const SET_TIMEZONE_VISIBILITY = `${ PREFIX_EVENTS_STORE }/SET_TIMEZONE_VISIBILITY`;
export const SET_TIMEZONE_LABEL = `${ PREFIX_EVENTS_STORE }/SET_TIMEZONE_LABEL`;

export const SET_DATE_INPUT_VISIBILITY = `${ PREFIX_EVENTS_STORE }/SET_DATE_INPUT_VISIBILITY`;

export const SET_DATETIME_BLOCK_EDITABLE_STATE = `${ PREFIX_EVENTS_STORE }/SET_DATETIME_BLOCK_EDITABLE_STATE`; // eslint-disable-line max-len
