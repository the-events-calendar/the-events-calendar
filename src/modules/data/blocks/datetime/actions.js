/**
 * Internal dependencies
 */
import * as types from './types';

export const setNaturalLanguageLabel = ( label, meta = {} ) => ( {
	type: types.SET_NATURAL_LANGUAGE_LABEL,
	payload: {
		label,
	},
	meta,
} );

export const setDateRange = ( payload, meta = {} ) => ( {
	type: types.SET_DATE_RANGE,
	payload,
	meta,
} );

export const setStartDateTime = ( start, meta = {} ) => ( {
	type: types.SET_START_DATE_TIME,
	payload: {
		start,
	},
	meta,
} );

export const setEndDateTime = ( end, meta = {} ) => ( {
	type: types.SET_END_DATE_TIME,
	payload: {
		end,
	},
	meta,
} );

export const setStartTime = ( start, meta = {} ) => ( {
	type: types.SET_START_TIME,
	payload: {
		start,
	},
	meta,
} );

export const setEndTime = ( end, meta = {} ) => ( {
	type: types.SET_END_TIME,
	payload: {
		end,
	},
	meta,
} );

export const setStartTimeInput = ( startTimeInput ) => ( {
	type: types.SET_START_TIME_INPUT,
	payload: {
		startTimeInput,
	},
} );

export const setEndTimeInput = ( endTimeInput ) => ( {
	type: types.SET_END_TIME_INPUT,
	payload: {
		endTimeInput,
	},
} );

export const setSeparatorDate = ( separator ) => ( {
	type: types.SET_SEPARATOR_DATE,
	payload: {
		separator,
	},
} );

export const setSeparatorTime = ( separator ) => ( {
	type: types.SET_SEPARATOR_TIME,
	payload: {
		separator,
	},
} );

export const setAllDay = ( allDay ) => ( {
	type: types.SET_ALL_DAY,
	payload: {
		allDay,
	},
} );

export const setMultiDay = ( multiDay, meta = {} ) => ( {
	type: types.SET_MULTI_DAY,
	payload: {
		multiDay,
	},
	meta,
} );

export const setTimeZone = ( timeZone ) => ( {
	type: types.SET_TIME_ZONE,
	payload: {
		timeZone,
	},
} );

export const setTimeZoneVisibility = ( show ) => ( {
	type: types.SET_TIMEZONE_VISIBILITY,
	payload: {
		show,
	},
} );

export const allowEdits = () => ( {
	type: types.SET_DATETIME_BLOCK_EDITABLE_STATE,
	payload: {
		isEditable: true,
	},
} );

export const disableEdits = () => ( {
	type: types.SET_DATETIME_BLOCK_EDITABLE_STATE,
	payload: {
		isEditable: false,
	},
} );
