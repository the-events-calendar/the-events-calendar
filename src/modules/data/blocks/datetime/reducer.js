/**
 * External dependencies
 */
import { __ } from '@wordpress/i18n';
import moment from 'moment/moment';

/**
 * Internal dependencies
 */
import {
	globals,
	date,
	moment as momentUtil,
	time,
} from '@moderntribe/common/utils';
import * as types from './types';

const { parseFormats, toTime } = momentUtil;

export const defaultStateToMetaMap = {
	start: '_EventStartDate',
	end: '_EventEndDate',
	dateTimeSeparator: '_EventDateTimeSeparator',
	timeRangeSeparator: '_EventTimeRangeSeparator',
	allDay: '_EventAllDay',
	timeZone: '_EventTimezone',
};

export const setInitialState = ( data ) => {
	const { meta } = data;

	Object.keys( defaultStateToMetaMap ).forEach( ( key ) => {
		const metaKey = defaultStateToMetaMap[ key ];
		if ( meta.hasOwnProperty( metaKey ) ) {
			DEFAULT_STATE[ key ] = meta[ metaKey ];
		}
	} );

	const { start, end } = DEFAULT_STATE;

	DEFAULT_STATE.startTimeInput = toTime( parseFormats( start ) );
	DEFAULT_STATE.endTimeInput = toTime( parseFormats( end ) );
	DEFAULT_STATE.naturalLanguageLabel = date.rangeToNaturalLanguage( start, end );
	DEFAULT_STATE.multiDay = ! momentUtil.isSameDay( momentUtil.toMoment( start ), momentUtil.toMoment( end ) );
};

const defaultStartTime = globals.defaultTimes().start ? globals.defaultTimes().start : '08:00:00';
const defaultEndTime = globals.defaultTimes().end ? globals.defaultTimes().end : '17:00:00';
const defaultStartTimeSeconds = time.toSeconds( defaultStartTime, time.TIME_FORMAT_HH_MM_SS );
const defaultEndTimeSeconds = time.toSeconds( defaultEndTime, time.TIME_FORMAT_HH_MM_SS );

export const defaultStartMoment = moment().startOf( 'day' ).seconds( defaultStartTimeSeconds );
export const defaultEndMoment = moment().startOf( 'day' ).seconds( defaultEndTimeSeconds );

const defaultStartDateTime = momentUtil.toDateTime( defaultStartMoment );
const defaultEndDateTime = momentUtil.toDateTime( defaultEndMoment );

export const DEFAULT_STATE = {
	start: defaultStartDateTime,
	end: defaultEndDateTime,
	startTimeInput: momentUtil.toTime( defaultStartMoment ),
	endTimeInput: momentUtil.toTime( defaultEndMoment ),
	naturalLanguageLabel: date.rangeToNaturalLanguage( defaultStartDateTime, defaultEndDateTime ),
	dateTimeSeparator: globals.settings().dateTimeSeparator
		? globals.settings().dateTimeSeparator
		: __( '@', 'the-events-calendar' ),
	timeRangeSeparator: globals.settings().timeRangeSeparator
		? globals.settings().timeRangeSeparator
		: __( '-', 'the-events-calendar' ),
	allDay: false,
	multiDay: false,
	timeZone: globals.timezone().timeZone ? globals.timezone().timeZone : date.FORMATS.TIMEZONE.string,
	showTimeZone: false,
	isEditable: true,
};

export default ( state = DEFAULT_STATE, action ) => {
	switch ( action.type ) {
		case types.SET_DATETIME_BLOCK_EDITABLE_STATE:
			return {
				...state,
				isEditable: action.payload.isEditable,
			};
		case types.SET_START_DATE_TIME:
			return {
				...state,
				start: action.payload.start,
			};
		case types.SET_END_DATE_TIME:
			return {
				...state,
				end: action.payload.end,
			};
		case types.SET_START_TIME_INPUT:
			return {
				...state,
				startTimeInput: action.payload.startTimeInput,
			};
		case types.SET_END_TIME_INPUT:
			return {
				...state,
				endTimeInput: action.payload.endTimeInput,
			};
		case types.SET_NATURAL_LANGUAGE_LABEL:
			return {
				...state,
				naturalLanguageLabel: action.payload.label,
			};
		case types.SET_ALL_DAY:
			return {
				...state,
				allDay: action.payload.allDay,
			};
		case types.SET_MULTI_DAY:
			return {
				...state,
				multiDay: action.payload.multiDay,
			};
		case types.SET_SEPARATOR_DATE:
			return {
				...state,
				dateTimeSeparator: action.payload.separator,
			};
		case types.SET_SEPARATOR_TIME:
			return {
				...state,
				timeRangeSeparator: action.payload.separator,
			};
		case types.SET_TIME_ZONE:
			return {
				...state,
				timeZone: action.payload.timeZone,
			};
		case types.SET_TIMEZONE_VISIBILITY:
			return {
				...state,
				showTimeZone: action.payload.show,
			};
		default:
			return state;
	}
};
