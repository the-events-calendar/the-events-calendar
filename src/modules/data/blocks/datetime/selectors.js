/**
 * External dependencies
 */
import { createSelector } from 'reselect';

export const datetimeSelector = ( state ) => state.events.blocks.datetime;

export const getStart = createSelector(
	[ datetimeSelector ],
	( datetime ) => datetime.start,
);

export const getEnd = createSelector(
	[ datetimeSelector ],
	( datetime ) => datetime.end,
);

export const getStartTimeInput = createSelector(
	[ datetimeSelector ],
	( datetime ) => datetime.startTimeInput,
);

export const getEndTimeInput = createSelector(
	[ datetimeSelector ],
	( datetime ) => datetime.endTimeInput,
);

export const getAllDay = createSelector(
	[ datetimeSelector ],
	( datetime ) => datetime.allDay,
);

export const getMultiDay = createSelector(
	[ datetimeSelector ],
	( datetime ) => datetime.multiDay,
);

export const getDateSeparator = createSelector(
	[ datetimeSelector ],
	( datetime ) => datetime.dateTimeSeparator,
);

export const getTimeSeparator = createSelector(
	[ datetimeSelector ],
	( datetime ) => datetime.timeRangeSeparator,
);

export const getTimeZone = createSelector(
	[ datetimeSelector ],
	( datetime ) => datetime.timeZone,
);

export const getTimeZoneVisibility = createSelector(
	[ datetimeSelector ],
	( datetime ) => datetime.showTimeZone,
);

export const getNaturalLanguageLabel = createSelector(
	[ datetimeSelector ],
	( datetime ) => datetime.naturalLanguageLabel,
);

export const isEditable = createSelector(
	[ datetimeSelector ],
	( datetime ) => datetime.isEditable,
);

export const getSameStartEnd = createSelector(
	[ datetimeSelector ],
	( datetime ) => datetime.start === datetime.end,
);
