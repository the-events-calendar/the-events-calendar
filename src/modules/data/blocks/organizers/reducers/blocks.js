/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import * as types from './../types';

export const byId = ( state = {}, action ) => {
	switch ( action.type ) {
		case types.ADD_BLOCK_ORGANIZER:
			return {
				...state,
				[ action.payload.id ]: action.payload.organizer,
			};
		case types.REMOVE_BLOCK_ORGANIZER:
			return Object.keys( state ).reduce( ( newState, id ) => {
				if ( id === action.payload.id ) {
					return newState;
				}

				return {
					...newState,
					[ id ]: state[ id ],
				};
			}, {} );
		default:
			return state;
	}
};

export const allIds = ( state = [], action ) => {
	switch ( action.type ) {
		case types.ADD_BLOCK_ORGANIZER:
			return [ ...state, action.payload.organizer ];
		case types.REMOVE_BLOCK_ORGANIZER:
			return state.filter( ( organizer ) => organizer !== action.payload.organizer );
		default:
			return state;
	}
};

export default combineReducers( {
	byId,
	allIds,
} );
