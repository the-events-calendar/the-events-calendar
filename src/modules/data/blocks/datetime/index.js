/**
 * Internal dependencies
 */
import * as reducer from './reducer';
import * as types from './types';
import * as actions from './actions';
import * as selectors from './selectors';
import sagas from './sagas';

export default reducer.default;
export {
	reducer,
	types,
	actions,
	selectors,
	sagas,
};
