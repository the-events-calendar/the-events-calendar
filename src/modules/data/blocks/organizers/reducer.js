/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import { classic, classicSetInitialState, blocks } from './reducers';

export const setInitialState = ( data ) => {
	classicSetInitialState( data );
};

export default combineReducers( {
	blocks,
	classic,
} );
