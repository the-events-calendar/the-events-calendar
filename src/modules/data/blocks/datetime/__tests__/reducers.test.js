/**
 * Internal dependencies
 */
import { actions } from '@moderntribe/events/data/blocks/datetime';
import reducer, {
	DEFAULT_STATE,
	defaultStateToMetaMap,
	setInitialState,
} from '@moderntribe/events/data/blocks/datetime/reducer';

jest.mock( 'moment', () => () => {
	const moment = require.requireActual( 'moment' );
	return moment( 'July 19, 2018 7:30 pm', 'MMMM D, Y h:mm a' );
} );

const data = {
	meta: {
		_EventStartDate: '2020-04-01 09:30:00',
		_EventEndDate: '2018-04-01 12:15:00',
		_EventDateTimeSeparator: 'at',
		_EventTimeRangeSeparator: 'to',
		_EventAllDay: false,
		_EventTimezone: 'UTC+8',
	},
};

describe( '[STORE] - Datetime reducer', () => {
	it( 'Should set the default state', () => {
		expect( reducer( undefined, {} ) ).toMatchSnapshot();
	} );

	it( 'Should set the start', () => {
		expect( reducer( DEFAULT_STATE, actions.setStartDateTime( 'June 5, 2018 5:00 pm' ) ) )
			.toMatchSnapshot();
	} );

	it( 'Should set the end', () => {
		expect( reducer( DEFAULT_STATE, actions.setEndDateTime( 'June 25, 2018 4:00 pm' ) ) ).toMatchSnapshot();
	} );

	it( 'Should set the start time input', () => {
		expect( reducer( DEFAULT_STATE, actions.setStartTimeInput( '18:00' ) ) ).toMatchSnapshot();
	} );

	it( 'Should set the end time input', () => {
		expect( reducer( DEFAULT_STATE, actions.setEndTimeInput( '18:00' ) ) ).toMatchSnapshot();
	} );

	it( 'Should set the separator time', () => {
		expect( reducer( DEFAULT_STATE, actions.setSeparatorTime( ' | ' ) ) ).toMatchSnapshot();
	} );

	it( 'Should set the separator date', () => {
		expect( reducer( DEFAULT_STATE, actions.setSeparatorDate( ' > ' ) ) ).toMatchSnapshot();
	} );

	it( 'Should set the timezone', () => {
		expect( reducer( DEFAULT_STATE, actions.setTimeZone( 'UTC' ) ) ).toMatchSnapshot();
	} );

	it( 'Should set the visibility of the timezone', () => {
		expect( reducer( DEFAULT_STATE, actions.setTimeZoneVisibility( true ) ) ).toMatchSnapshot();
		expect( reducer( DEFAULT_STATE, actions.setTimeZoneVisibility( false ) ) ).toMatchSnapshot();
	} );

	it( 'Should set the all day', () => {
		expect( reducer( DEFAULT_STATE, actions.setAllDay( true ) ) ).toMatchSnapshot();
		expect( reducer( DEFAULT_STATE, actions.setAllDay( false ) ) ).toMatchSnapshot();
	} );

	it( 'Should set the multi day', () => {
		expect( reducer( DEFAULT_STATE, actions.setMultiDay( true ) ) ).toMatchSnapshot();
		expect( reducer( DEFAULT_STATE, actions.setMultiDay( false ) ) ).toMatchSnapshot();
	} );

	it( 'Should set the natural language label', () => {
		expect( reducer( DEFAULT_STATE, actions.setNaturalLanguageLabel( '2 weeks from now' ) ) )
			.toMatchSnapshot();
	} );

	it( 'Should set the editability', () => {
		expect( reducer( DEFAULT_STATE, actions.allowEdits() ) ).toMatchSnapshot();
		expect( reducer( DEFAULT_STATE, actions.disableEdits() ) ).toMatchSnapshot();
	} );

	it( 'Should return the default state to meta map', () => {
		expect( defaultStateToMetaMap ).toMatchSnapshot();
	} );

	it( 'Should set the initial state', () => {
		setInitialState( data );
		expect( DEFAULT_STATE ).toMatchSnapshot();
	} );
} );
