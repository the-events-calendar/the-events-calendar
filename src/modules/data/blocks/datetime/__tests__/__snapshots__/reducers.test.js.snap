// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[STORE] - Datetime reducer Should return the default state to meta map 1`] = `
Object {
  "allDay": "_EventAllDay",
  "dateTimeSeparator": "_EventDateTimeSeparator",
  "end": "_EventEndDate",
  "start": "_EventStartDate",
  "timeRangeSeparator": "_EventTimeRangeSeparator",
  "timeZone": "_EventTimezone",
}
`;

exports[`[STORE] - Datetime reducer Should set the all day 1`] = `
Object {
  "allDay": true,
  "dateTimeSeparator": "@",
  "end": "2018-07-19 17:00:00",
  "endTimeInput": "5:00 pm",
  "isEditable": true,
  "multiDay": false,
  "naturalLanguageLabel": "July 19 2018 at 8:00 am - 5:00 pm",
  "showTimeZone": false,
  "start": "2018-07-19 08:00:00",
  "startTimeInput": "8:00 am",
  "timeRangeSeparator": "-",
  "timeZone": "UTC",
}
`;

exports[`[STORE] - Datetime reducer Should set the all day 2`] = `
Object {
  "allDay": false,
  "dateTimeSeparator": "@",
  "end": "2018-07-19 17:00:00",
  "endTimeInput": "5:00 pm",
  "isEditable": true,
  "multiDay": false,
  "naturalLanguageLabel": "July 19 2018 at 8:00 am - 5:00 pm",
  "showTimeZone": false,
  "start": "2018-07-19 08:00:00",
  "startTimeInput": "8:00 am",
  "timeRangeSeparator": "-",
  "timeZone": "UTC",
}
`;

exports[`[STORE] - Datetime reducer Should set the default state 1`] = `
Object {
  "allDay": false,
  "dateTimeSeparator": "@",
  "end": "2018-07-19 17:00:00",
  "endTimeInput": "5:00 pm",
  "isEditable": true,
  "multiDay": false,
  "naturalLanguageLabel": "July 19 2018 at 8:00 am - 5:00 pm",
  "showTimeZone": false,
  "start": "2018-07-19 08:00:00",
  "startTimeInput": "8:00 am",
  "timeRangeSeparator": "-",
  "timeZone": "UTC",
}
`;

exports[`[STORE] - Datetime reducer Should set the editability 1`] = `
Object {
  "allDay": false,
  "dateTimeSeparator": "@",
  "end": "2018-07-19 17:00:00",
  "endTimeInput": "5:00 pm",
  "isEditable": true,
  "multiDay": false,
  "naturalLanguageLabel": "July 19 2018 at 8:00 am - 5:00 pm",
  "showTimeZone": false,
  "start": "2018-07-19 08:00:00",
  "startTimeInput": "8:00 am",
  "timeRangeSeparator": "-",
  "timeZone": "UTC",
}
`;

exports[`[STORE] - Datetime reducer Should set the editability 2`] = `
Object {
  "allDay": false,
  "dateTimeSeparator": "@",
  "end": "2018-07-19 17:00:00",
  "endTimeInput": "5:00 pm",
  "isEditable": false,
  "multiDay": false,
  "naturalLanguageLabel": "July 19 2018 at 8:00 am - 5:00 pm",
  "showTimeZone": false,
  "start": "2018-07-19 08:00:00",
  "startTimeInput": "8:00 am",
  "timeRangeSeparator": "-",
  "timeZone": "UTC",
}
`;

exports[`[STORE] - Datetime reducer Should set the end 1`] = `
Object {
  "allDay": false,
  "dateTimeSeparator": "@",
  "end": "June 25, 2018 4:00 pm",
  "endTimeInput": "5:00 pm",
  "isEditable": true,
  "multiDay": false,
  "naturalLanguageLabel": "July 19 2018 at 8:00 am - 5:00 pm",
  "showTimeZone": false,
  "start": "2018-07-19 08:00:00",
  "startTimeInput": "8:00 am",
  "timeRangeSeparator": "-",
  "timeZone": "UTC",
}
`;

exports[`[STORE] - Datetime reducer Should set the end time input 1`] = `
Object {
  "allDay": false,
  "dateTimeSeparator": "@",
  "end": "2018-07-19 17:00:00",
  "endTimeInput": "18:00",
  "isEditable": true,
  "multiDay": false,
  "naturalLanguageLabel": "July 19 2018 at 8:00 am - 5:00 pm",
  "showTimeZone": false,
  "start": "2018-07-19 08:00:00",
  "startTimeInput": "8:00 am",
  "timeRangeSeparator": "-",
  "timeZone": "UTC",
}
`;

exports[`[STORE] - Datetime reducer Should set the initial state 1`] = `
Object {
  "allDay": false,
  "dateTimeSeparator": "at",
  "end": "2018-04-01 12:15:00",
  "endTimeInput": "12:15 pm",
  "isEditable": true,
  "multiDay": true,
  "naturalLanguageLabel": "April 1 2020 at 9:30 am - April 1 2018 at 12:15 pm",
  "showTimeZone": false,
  "start": "2020-04-01 09:30:00",
  "startTimeInput": "9:30 am",
  "timeRangeSeparator": "to",
  "timeZone": "UTC+8",
}
`;

exports[`[STORE] - Datetime reducer Should set the multi day 1`] = `
Object {
  "allDay": false,
  "dateTimeSeparator": "@",
  "end": "2018-07-19 17:00:00",
  "endTimeInput": "5:00 pm",
  "isEditable": true,
  "multiDay": true,
  "naturalLanguageLabel": "July 19 2018 at 8:00 am - 5:00 pm",
  "showTimeZone": false,
  "start": "2018-07-19 08:00:00",
  "startTimeInput": "8:00 am",
  "timeRangeSeparator": "-",
  "timeZone": "UTC",
}
`;

exports[`[STORE] - Datetime reducer Should set the multi day 2`] = `
Object {
  "allDay": false,
  "dateTimeSeparator": "@",
  "end": "2018-07-19 17:00:00",
  "endTimeInput": "5:00 pm",
  "isEditable": true,
  "multiDay": false,
  "naturalLanguageLabel": "July 19 2018 at 8:00 am - 5:00 pm",
  "showTimeZone": false,
  "start": "2018-07-19 08:00:00",
  "startTimeInput": "8:00 am",
  "timeRangeSeparator": "-",
  "timeZone": "UTC",
}
`;

exports[`[STORE] - Datetime reducer Should set the natural language label 1`] = `
Object {
  "allDay": false,
  "dateTimeSeparator": "@",
  "end": "2018-07-19 17:00:00",
  "endTimeInput": "5:00 pm",
  "isEditable": true,
  "multiDay": false,
  "naturalLanguageLabel": "2 weeks from now",
  "showTimeZone": false,
  "start": "2018-07-19 08:00:00",
  "startTimeInput": "8:00 am",
  "timeRangeSeparator": "-",
  "timeZone": "UTC",
}
`;

exports[`[STORE] - Datetime reducer Should set the separator date 1`] = `
Object {
  "allDay": false,
  "dateTimeSeparator": " > ",
  "end": "2018-07-19 17:00:00",
  "endTimeInput": "5:00 pm",
  "isEditable": true,
  "multiDay": false,
  "naturalLanguageLabel": "July 19 2018 at 8:00 am - 5:00 pm",
  "showTimeZone": false,
  "start": "2018-07-19 08:00:00",
  "startTimeInput": "8:00 am",
  "timeRangeSeparator": "-",
  "timeZone": "UTC",
}
`;

exports[`[STORE] - Datetime reducer Should set the separator time 1`] = `
Object {
  "allDay": false,
  "dateTimeSeparator": "@",
  "end": "2018-07-19 17:00:00",
  "endTimeInput": "5:00 pm",
  "isEditable": true,
  "multiDay": false,
  "naturalLanguageLabel": "July 19 2018 at 8:00 am - 5:00 pm",
  "showTimeZone": false,
  "start": "2018-07-19 08:00:00",
  "startTimeInput": "8:00 am",
  "timeRangeSeparator": " | ",
  "timeZone": "UTC",
}
`;

exports[`[STORE] - Datetime reducer Should set the start 1`] = `
Object {
  "allDay": false,
  "dateTimeSeparator": "@",
  "end": "2018-07-19 17:00:00",
  "endTimeInput": "5:00 pm",
  "isEditable": true,
  "multiDay": false,
  "naturalLanguageLabel": "July 19 2018 at 8:00 am - 5:00 pm",
  "showTimeZone": false,
  "start": "June 5, 2018 5:00 pm",
  "startTimeInput": "8:00 am",
  "timeRangeSeparator": "-",
  "timeZone": "UTC",
}
`;

exports[`[STORE] - Datetime reducer Should set the start time input 1`] = `
Object {
  "allDay": false,
  "dateTimeSeparator": "@",
  "end": "2018-07-19 17:00:00",
  "endTimeInput": "5:00 pm",
  "isEditable": true,
  "multiDay": false,
  "naturalLanguageLabel": "July 19 2018 at 8:00 am - 5:00 pm",
  "showTimeZone": false,
  "start": "2018-07-19 08:00:00",
  "startTimeInput": "18:00",
  "timeRangeSeparator": "-",
  "timeZone": "UTC",
}
`;

exports[`[STORE] - Datetime reducer Should set the timezone 1`] = `
Object {
  "allDay": false,
  "dateTimeSeparator": "@",
  "end": "2018-07-19 17:00:00",
  "endTimeInput": "5:00 pm",
  "isEditable": true,
  "multiDay": false,
  "naturalLanguageLabel": "July 19 2018 at 8:00 am - 5:00 pm",
  "showTimeZone": false,
  "start": "2018-07-19 08:00:00",
  "startTimeInput": "8:00 am",
  "timeRangeSeparator": "-",
  "timeZone": "UTC",
}
`;

exports[`[STORE] - Datetime reducer Should set the visibility of the timezone 1`] = `
Object {
  "allDay": false,
  "dateTimeSeparator": "@",
  "end": "2018-07-19 17:00:00",
  "endTimeInput": "5:00 pm",
  "isEditable": true,
  "multiDay": false,
  "naturalLanguageLabel": "July 19 2018 at 8:00 am - 5:00 pm",
  "showTimeZone": true,
  "start": "2018-07-19 08:00:00",
  "startTimeInput": "8:00 am",
  "timeRangeSeparator": "-",
  "timeZone": "UTC",
}
`;

exports[`[STORE] - Datetime reducer Should set the visibility of the timezone 2`] = `
Object {
  "allDay": false,
  "dateTimeSeparator": "@",
  "end": "2018-07-19 17:00:00",
  "endTimeInput": "5:00 pm",
  "isEditable": true,
  "multiDay": false,
  "naturalLanguageLabel": "July 19 2018 at 8:00 am - 5:00 pm",
  "showTimeZone": false,
  "start": "2018-07-19 08:00:00",
  "startTimeInput": "8:00 am",
  "timeRangeSeparator": "-",
  "timeZone": "UTC",
}
`;
